.github/workflows/release.yaml

```
@@ -72,17 +72,26 @@ jobs:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      # ensure docs are always deployed after merge of changeset PR
+      - name: Set var SHOULD_DEPLOY_DOCS
+        run: |
+          echo "SHOULD_DEPLOY_DOCS=$( \
+            [[ '${{ steps.changesets.outputs.published }}' == 'true' ]] || \
+            git --no-pager log -1 --pretty=%B | grep -q '^ci(changesets):' \
+            && echo 'true' || echo 'false' \
+          )" >> $GITHUB_ENV
+
```

Maintaining an embedded multiline shell script with special line-break formatting such as this can be tricky. I think it'd be simpler to go with two steps for increased clarity:

```suggestion
      - name: Get the last commit message and set env vars
        run: |
          echo LAST_COMMIT_MSG=$(git --no-pager log -1 --pretty=%B) >> $GITHUB_ENV
​          echo SHOULD_DEPLOY_DOCS=false >> $GITHUB_ENV

      - name: Decides if Docs should be deployed
        if: >-
          ${{
            steps.changesets.outputs.published == 'true' ||
            startsWith(env.LAST_COMMIT_MSG, 'ci(changesets):'
          }}
        run: echo SHOULD_DEPLOY_DOCS=true >> $GITHUB_ENV

```